src/main/java/org/apache/platypus/server/luceneserver/QueryNodeMapper.java

```diff
@@ -106,6 +115,40 @@ private DisjunctionMaxQuery getDisjunctionMaxQuery(org.apache.platypus.server.gr
         return new DisjunctionMaxQuery(disjuncts, disjunctionMaxQuery.getTieBreakerMultiplier());
     }
 
+    private Query getMatchQuery(MatchQuery matchQuery, IndexState state) {
+        Analyzer analyzer = isAnalyzerDefined(matchQuery.getAnalyzer())
+                ? getAnalyzer(matchQuery.getAnalyzer())
+                : state.searchAnalyzer;
+
+        MultiFieldQueryParser multiFieldQueryParser = new MultiFieldQueryParser(null, analyzer);
+
+        // This created query will be TermQuery if only one token is found after analysis, otherwise BooleanQuery
+        Query query = multiFieldQueryParser.createBooleanQuery(matchQuery.getField(), matchQuery.getQuery(), occurMapping.get(matchQuery.getOperator()));
```

Comment:
Seems like the base class QueryBuilder throws exception for any clauses other than MUST and SHOULD. What do you think about making a more restrictive proto clause with only those two options. Might be easier on the user to not worry about RuntimeError due to invalid clause.

```
    if (operator != BooleanClause.Occur.SHOULD && operator != BooleanClause.Occur.MUST) {
      throw new IllegalArgumentException("invalid operator: only SHOULD or MUST are allowed");
    }
```

Reply:
Added MatchOperator with only MUST and SHOULD.